src/container/Schedules/utilities.ts

```diff
@@ -119,3 +120,79 @@ export const memoizeData = (data: any) => {
   const makeDataLevel = () => data;
   return makeDataLevel();
 };
+
+// autocomplete cell functions
+export function extractRoleIds(teams): number[] {
+  return teams.map((team) => team.id);
+}
+
+export function getRoleOptionLabel(option, dataSet) {
+  const filteredData = dataSet.filter((role) => role.id === option)[0];
+  if (filteredData) return `${filteredData.name}`;
+  return '';
+}
+
+export function getUserOptionLabel(option: number, dataSet) {
+  const filteredData = dataSet.filter((user) => user.userId === option)[0];
+  if (filteredData) return `${filteredData.firstName} ${filteredData.lastName}`;
+  else {
+    // this is just to prevent errors for now
+    return '';
+  }
+}
+
+export function extractTeammateIds(teammates): number[] {
+  return teammates.map((teammate) => teammate.userId);
+}
+
+export const teammates = (data, roleId: number, churchId) => {
+  const filteredTeammates = data.users.filter((user) =>
+    user.teams.some((team) => team.id === roleId),
+  );
+  filteredTeammates.push(blankTeammate(churchId)); // adds blankteammate to list incase they wish to leave it blank
+  return filteredTeammates;
+};
+
+export const blankTeammate = (churchId: number): UsersDataInterface => {
+  return {
+    userId: -1,
+    firstName: '',
+    lastName: '',
+    email: '',
+    church: { name: '' },
+    churchId: churchId,
+    disabled: false,
+    teams: [],
+  };
+};
+
+const createBlankTask = (seedFx) => {
+  return {
+    taskId: seedFx(),
+    userId: seedFx(),
+  };
+};
+
+const createBlankEventCells = (cellLength: number, seedFx) => {
+  const taskCells: any = [{}, {}]; // data from these cells aren't actually being used, are just placeholders for rendering
+  for (let i = 2; i < cellLength; i++) {
+    // 2 (the columns for time/duty) is hardcoded here
```

Comment:
Explanatory variable to replace comment and magic number:
```suggestion
  const afterTimeAndDutyColumns = 2;
  for (let i = afterTimeAndDutyColumns; i < cellLength; i++) {
```